### ISSUE TITLE
Is there any plan to support auto completion for wxlua?

### ISSUE THREAD

**Issue by louislau86** [pkulchenko/ZeroBraneStudio#792]
hi, i am a zerobranestudio user, thanks for your good job!
But for wxLua development, it is not so convenient. So, is there any plan to support auto-completion or hint 
for wxLua? 

Thank you

**Comment by jjvbsag** [comment]
👍 As we will start with some big wxLua based code soon, this has my plus-one.
Would it be possible to generate wxlua api files from the wxlua project by script? 
@pkulchenko when I'm back home, i could give it a try, unless you have another idea or a quicker solution.

**Comment by louislau86** [comment]
@jjvbsag  I am happy to meet with a experienced wxLua user! I' m a freshman for it.  I'm interested in which ui design tool have you used and is there any good idea for wxLua to solve the problem that ui is not so smooth when there is a long Time-consuming operation? Because lua is not support multi-thread and i'm worry about the effects of the coroutine. And more, is wxLua a good choose for a big project? Thank you and expect your advises!

**Comment by jjvbsag** [comment]
@louislau86 (not sure if this is the right place to discuss, but at least I want to give _some_ reply)

"starting big wxLua bases project soon" does not say, I already _am_ an experienced wxLua user. I did my first steps decades ago with Perl/Tk, then some years ago changed to Lua/Iup. Now using ZBS for some times, it makes more sense for me to use wxLua, also because Paul is doing a great job (🥇) and ZBS is quite tight bundled with wxLua.

Anyway, Yes, I believe big projects _can_ be done with Lua/wxLua. And regarding the "long Time-consuming" operations: In IUP I always used a combination of coroutine and IupIdle, this should be possible in wxLua too.

**Comment by louislau86** [comment]
@jjvbsag  thanks for your reply!
But for @pkulchenko i'm sorry to have a discussion here,  maybe it is more appropriate to discuss the 1st question~

**Comment by pkulchenko** [comment]
> Would it be possible to generate wxlua api files from the wxlua project by script?

Yes, it may be doable; in fact, I've recently added a [script](https://github.com/pkulchenko/wxlua/blob/wxwidgets311/wxLua/bindings/any-bind-sync.lua) that is used to update wxlua binding from wxwidgets descriptions and likely some of it can be used to generate wxlua API descriptions as well.

> i'm sorry to have a discussion here, maybe it is more appropriate to discuss the 1st question.

@louislau86, no worries; this is a good discussion to have. There is a stub for wxwidgets API (in api/lua/wxwidgets.lua), but it only knows about wxwidgets values/functions in the global namespace, but not about their methods or anything else. It would be nice to have scripted API description for wxlua functions.
